### ISSUE TITLE
Frame Throughput (FT)

### ISSUE THREAD

**Issue by digitarald** [Mozilla-TWQA/Hasal#543]
Spin off from #518

1. Wait until the page is loaded and visually complete
2. Determine when the page animates or scrolls (deferred by Hasal triggering scrolling)
  - *Is-it-Animating* heuristic is basically just checking if the video frame updated several times per second
3. Collect frame durations for the animation period as histogram to determine the *frequency and duration dropped frames
4. Aggregate dropped frame measures as done in [bug 1338347](https://bugzilla.mozilla.org/show_bug.cgi?id=1338347) for comparison

**Comment by digitarald** [comment]
@MikeLien did I capture the flow correctly for Hasal? Please change the spec to your needs.

**Comment by MikeLien** [comment]
> Is-it-Animating heuristic is basically just checking if the video frame updated several times per second
> Collect frame durations for the animation period as histogram to determine the *frequency and duration dropped frames

We'll restrict script's start point at visual complete of page load. For currently implementation, scrolling measurement and input latency of typing have the same start points but with different measurement mechanism. The measurement mechanism of input latency of typing is video frame updated latency under a higher fps of screen recording environment. Scrolling measurement has different situation, for getting dropped frame of scrolling action from video of screen recording, we will need a golden sample which is a non-dropped frame data to compare with each run then we can measure the gap between them.

**Comment by digitarald** [comment]
> visual complete of page load

What does that mean explicitly? Is it still observing the the tab spinner?

**Comment by MikeLien** [comment]
In general, yes, but it depends on the case design. We can also add customized anchor on screen to set the start point. For example, to avoid ambiguous start point from visual complete of page load to typing event fired in input latency of typing case during video analysis, I may shift the start point by pasting a string on screen and start interactions.

**Comment by digitarald** [comment]
> In general, yes, but it depends on the case design.

Could we consider alternatives like monitor the network for being idle or close (3s with just one connection open)? We saw in Google Spreadsheets that a lot of app components loaded lazily, like the avatars and the Google drive integration (Move To)

**Comment by bobbyc** [comment]
@digitarald At your step 3) in the first comment, do you mean collect frame data from profiler data? or are you saying we should analyze recording to find drop frame and checkboarding frame?

**Comment by digitarald** [comment]
@bobbyc my assumption was to get that data from the video. It should be possible to get it from the profiles as well though the data has to be mirrored for Chrome's traces

**Comment by bobbyc** [comment]
@digitarald do you expect we use the same formula in [Bug 1338347](https://bugzilla.mozilla.org/show_bug.cgi?id=1338347) to calculate frame ratio?

Frame Ratio:
  Actual Number of Frames / Expected Number of Frames
  0-1 where 1 equals no dropped frames (0.5 half of frames dropped, etc.)

Long Frame:
  99th percentile frame length (in ms)

**Comment by digitarald** [comment]
@bobbyc collecting the full histograms for frame drops would be step 1. Then we can try the summary that I proposed with median and 99th percentile and analyse how well it reflects the distribution.

**Comment by bobbyc** [comment]
> @bobbyc <https://github.com/bobbyc> collecting the full histograms for
> frame drops would be step 1. Then we can try the summary that I proposed
> with median and 99th percentile and analyse how well it reflects the
> distribution.


​sounds like a plan.​ thanks.

**Comment by digitarald** [comment]
@bobbyc just ton confirm, are we still on target for collecting the frame time histogram?

**Comment by MikeLien** [comment]
I'm prototyping to get these values and what I already get now are:

- Frame Ratio: `Actual Number of Frames / Expected Number of Frames`
- Long Frame: `99th percentile frame length (in ms)`

I'd like to know how's the frame time histogram you expect? Is it how along each frame appear on screen?

**Comment by digitarald** [comment]
Please see discussion in https://bugzilla.mozilla.org/show_bug.cgi?id=1338347 on making the measure vsync independent as frame drop ratio. Apart from the simplification to only report max and mean to summarize the histogram, we do something similar here?

**Comment by digitarald** [comment]
@MikeLien, forgot to ping, see above

**Comment by MikeLien** [comment]
Here are our prototype's [results](https://drive.google.com/open?id=0BwkEhia_D6l_QkFKa1pqRTRISHM) for firefox and chrome with gdoc scrolling under `one scroll event` and `scroll to end`.

From json file's raw data you can find the expected total frames of firefox are greater than chrome's. The reason is different display behaviors between both browsers. For example, the `one scroll event` will have same target movement from both browsers but firefox behave many slight scene changes and chrome behave a heavy change instead. For more detailed information you can refer to the video or images in the result folder.

**Comment by bobbyc** [comment]
Raw data analysis: https://gist.github.com/bobbyc/b7d2f333de37603d4624abdbb8ba8154

**Comment by digitarald** [comment]
> Here are our prototype's results for firefox and chrome with gdoc scrolling under one scroll event and scroll to end cases.

How does `scroll to end` work?

> or example, the one scroll event will have same movement target from both browsers but firefox behave many slight scene changes and chrome behave a heavy change instead.

Looks like Chrome doesn't have smooth scrolling and directly jumps. How do we capture any frames for Chrome when the is no animation whatsoever?

Can we simulate two finger touchpad scroll?

**Comment by MikeLien** [comment]
>> Here are our prototype's results for firefox and chrome with gdoc scrolling under one scroll event and scroll to end cases.

>How does `scroll to end` work?
Sikuli supports to send many scroll events in one command. As we test, sending 32 scroll events can get such case's doc to the end.

>> or example, the one scroll event will have same movement target from both browsers but firefox behave many slight scene changes and chrome behave a heavy change instead.

>Looks like Chrome doesn't have smooth scrolling and directly jumps. How do we capture any frames for Chrome when the is no animation whatsoever?

>Can we simulate two finger touchpad scroll?
This case is just `one scroll event`. All frames come from screen capture's video result with 60fps as its capture setting.

**Comment by MikeLien** [comment]
BTW, there's one thing for your reference, the span of scene change from one scroll event will be affected by OS's scroll setting. That can be varied and come out different results. If we need to get frame throughput regularly then the scroll settings from OS should be one variable.

**Comment by jeanygong** [comment]
What is the status on the frame time histogram?

**Comment by digitarald** [comment]
> Sikuli supports to send many scroll events in one command. We only need to make sure after sending events finished, if it's already one the end of doc. Then video analysis will try to find the last scene change.

Are we using the same scroll technique for `one` vs `end` scroll?

> I think the touchpad and mouse scroll use the same triggered event, is it?

Not for APZ, they are handled differently.

For scroll test cases, could we add the input latency metric and only measure the FT for while the browser actually animates?

**Comment by bobbyc** [comment]
Hi @jeanygong @digitarald

The histogram is in previous comment. Paste this again. 
Raw data analysis: https://gist.github.com/bobbyc/b7d2f333de37603d4624abdbb8ba8154

**Comment by bobbyc** [comment]
Previous report was a single run histogram. The 30times run data will be coming today. FT testing is a time comsuming tests. Sorry for late.

**Comment by MikeLien** [comment]
> Are we using the same scroll technique for one vs end scroll?

Yes, we use the same scroll command and only change the number of scroll event sending in this command.

> For scroll test cases, could we add the input latency metric and only measure the FT for while the browser actually animates?

The measurement period of input latency is from event log display until first expected display change and I dont't think this is proper to apply on FT measurement since this doesn't take animation start point as first frame.
The prototype's FT result already based on the period of displayed animation from browser to claculate, for more detailed what we got please refer to the video or image results from above comment.

**Comment by bobbyc** [comment]
@digitarald @jeanygong 

FT chrome scroll to end (30times): https://gist.github.com/bobbyc/b4d6aabea66754e6167e50e4ac8cebe5

FT Firefox scroll to end (30times): https://gist.github.com/bobbyc/423ffeac505c00bf158e8820b32f0f9a

**Comment by digitarald** [comment]
> … I don't think this is proper to apply on FT measurement since this doesn't take animation start point as first frame.

@MikeLien can you explain some more why this would not work. Since we know when we fire the scroll event and we know when the first frame animates to scrolls; we can tell the latency between them. While it sounds easy to break down the metrics, I guess there is some difference in how FT and IL tests are currently setup as IL depends on the log as reference. Can they both be reported at the same time or should we split up the test?

> Reports:

@bobbyc looks like both gists have the same data ;) . The data for Chrome looks unexpectedly low; can you share videos to understand what's happening there?

**Comment by bobbyc** [comment]
@digitarald Same data sets, but different plot results. I just want to emphasize Chrome and Firefox have different behaviors. As observations, Chrome tends to skip frames during fast scroll and give user acceptable results; so you can see the longest frame is about 341ms long (in average). Firefox tends to generate every frame; so you can see the longest frame is about 26ms long (in average)

**Comment by bobbyc** [comment]
@digitarald  You could find videos here: https://drive.google.com/drive/folders/0BwkEhia_D6l_djIyejRDamg1RFU. 

>@bobbyc looks like both gists have the same data ;) . The data for Chrome looks unexpectedly low; can you share videos to understand what's happening there?

https://drive.google.com/drive/folders/0BwkEhia_D6l_WUxWdUZMTEs2Zkk

**Comment by digitarald** [comment]
> @digitarald You could find videos here: https://drive.google.com/drive/folders/0BwkEhia_D6l_djIyejRDamg1RFU.

@bobbyc If [this video](https://drive.google.com/open?id=0BwkEhia_D6l_MjhKRTFQSXItRE0) shows the test for Chrome and the 1-scroll case, there is no smooth scroll.

> https://drive.google.com/drive/folders/0BwkEhia_D6l_WUxWdUZMTEs2Zkk

The [scroll end video](https://drive.google.com/drive/folders/0BwkEhia_D6l_WUxWdUZMTEs2Zkk) shows that Chrome is also not smooth scrolling but jumping a 1-scroll distance for each scroll input. These benchmarks can not be compared until we can find the right input method for Chrome to trigger smooth scroll.

**Comment by MikeLien** [comment]
> can you explain some more why this would not work. Since we know when we fire the scroll event and we know when the first frame animates to scrolls; we can tell the latency between them. While it sounds easy to break down the metrics, I guess there is some difference in how FT and IL tests are currently setup as IL depends on the log as reference. Can they both be reported at the same time or should we split up the test?

I'm not saying we cannot use input latency's measurement method to test frame throughput case, what I mean is that if we take such result then we will get log shown time as start point instead of animation's first scene chang time. This should be considered to get precisely value.
For now, input latency and frame throughput measurement will be two different cases. We design and run these two kind of cases separately but I think as you mentioned one case report two different indexes is doable and we can try to make it happen.

**Comment by MikeLien** [comment]
>> @digitarald You could find videos here: https://drive.google.com/drive/folders/0BwkEhia_D6l_djIyejRDamg1RFU.

> @bobbyc If [this video](https://drive.google.com/open?id=0BwkEhia_D6l_MjhKRTFQSXItRE0) shows the test for Chrome and the 1-scroll case, there is no smooth scroll.

>> https://drive.google.com/drive/folders/0BwkEhia_D6l_WUxWdUZMTEs2Zkk

> The [scroll end video](https://drive.google.com/drive/folders/0BwkEhia_D6l_WUxWdUZMTEs2Zkk) shows that Chrome is also not smooth scrolling but jumping a 1-scroll distance for each scroll input. These benchmarks can not be compared until we can find the right input method for Chrome to trigger smooth scroll.

I think Bobby is on his flight now and I can reply first. All the one time and 30 times results came from Windows with Firefox nightly(02-23) and Chrome(57.0.2987.110). On Windows, we use mouse scroll's default setting(one scroll will move three lines) to run our prototype case but different browsers behave different ways. As I mentioned above, the `one scroll event` have the same movement target on Firefox and Chrome but Firefox behave many slight scene changes to make it looks smooth and draw many frames while Chrome just draw the movement target frame directly. Even executed by real user with 1-scroll will get the same result.

**Comment by digitarald** [comment]
@ShakoHo @MikeLien we need to check if Firefox uses hardware rendering. Can we get the raw data from about:support for the test machines?

**Comment by ShakoHo** [comment]
@digitarald 
Please check the json below:

Windows Machine
```json
{
  "application": {
    "name": "Firefox",
    "osVersion": "Windows_NT 6.1",
    "version": "55.0a1",
    "buildID": "20170322030208",
    "userAgent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:55.0) Gecko/20100101 Firefox/55.0",
    "safeMode": false,
    "updateChannel": "nightly",
    "supportURL": "https://support.mozilla.org/1/firefox/55.0a1/WINNT/en-US/",
    "numTotalWindows": 1,
    "numRemoteWindows": 1,
    "remoteAutoStart": true,
    "autoStartStatus": 1,
    "keyGoogleFound": true,
    "keyMozillaFound": true
  },
  "modifiedPreferences": {
    "browser.cache.disk.hashstats_reported": 1,
    "browser.cache.frecency_experiment": 4,
    "browser.cache.disk.smart_size.first_run": false,
    "browser.cache.disk.smart_size.use_old_max": false,
    "browser.cache.disk.capacity": 51200,
    "browser.cache.disk.filesystem_reported": 1,
    "browser.cache.disk.smart_size_cached_value": 358400,
    "browser.download.importedFromSqlite": true,
    "browser.download.manager.alertOnEXEOpen": true,
    "browser.places.importBookmarksHTML": false,
    "browser.places.smartBookmarksVersion": 8,
    "browser.sessionstore.upgradeBackup.latestBuildID": "20170322030208",
    "browser.startup.homepage_override.buildID": "20170322030208",
    "browser.startup.homepage_override.mstone": "55.0a1",
    "browser.urlbar.lastSuggestionsPromptDate": 20161110,
    "browser.urlbar.daysBeforeHidingSuggestionsPrompt": 0,
    "browser.urlbar.usepreloadedtopurls.enabled": false,
    "dom.apps.lastUpdate.buildID": "20160922113459",
    "dom.apps.reset-permissions": true,
    "dom.gamepad.extensions.enabled": true,
    "dom.apps.lastUpdate.mstone": "49.0.1",
    "dom.mozApps.used": true,
    "extensions.lastAppVersion": "55.0a1",
    "gfx.crash-guard.status.d3d11layers": 2,
    "gfx.crash-guard.d3d11layers.appVersion": "49.0.1",
    "gfx.driver-init.feature-d3d11": true,
    "gfx.crash-guard.d3d11layers.feature-d3d11": true,
    "gfx.driver-init.deviceID": "0x0f00",
    "gfx.driver-init.driverVersion": "9.18.13.2049",
    "gfx.driver-init.status": 2,
    "gfx.driver-init.appVersion": "42.0",
    "gfx.crash-guard.d3d11layers.feature-d2d": true,
    "gfx.crash-guard.d3d11layers.driverVersion": "9.18.13.2049",
    "gfx.direct3d.last_used_feature_level_idx": 0,
    "gfx.driver-init.feature-d2d": true,
    "gfx.crash-guard.status.d3d9video": 2,
    "gfx.crash-guard.d3d11layers.deviceID": "0x0f00",
    "javascript.options.asyncstack": false,
    "media.gmp.storage.version.observed": 1,
    "media.gmp-eme-adobe.lastUpdate": 1464597454,
    "media.gmp-manager.buildID": "20170322030208",
    "media.benchmark.vp9.versioncheck": 2,
    "media.gmp-gmpopenh264.lastUpdate": 1490274749,
    "media.gmp-gmpopenh264.version": "1.6",
    "media.gmp-eme-adobe.abi": "x86-msvc-x64",
    "media.benchmark.vp9.fps": 194,
    "media.hardware-video-decoding.failed": false,
    "media.gmp-eme-adobe.version": "17",
    "media.gmp-gmpopenh264.abi": "x86_64-msvc-x64",
    "media.gmp-manager.lastCheck": 1490274748,
    "media.gmp-widevinecdm.lastUpdate": 1490274750,
    "media.gmp-widevinecdm.abi": "x86_64-msvc-x64",
    "media.gmp-widevinecdm.version": "1.4.8.903",
    "network.cookie.prefsMigrated": true,
    "network.predictor.cleaned-up": true,
    "places.history.expiration.transient_current_max_pages": 59,
    "places.database.lastMaintenance": 1486014066,
    "plugin.importedState": true,
    "plugin.disable_full_page_plugin_for_types": "application/pdf",
    "plugin.state.flash": 0,
    "privacy.sanitize.timeSpan": 0,
    "privacy.sanitize.migrateClearSavedPwdsOnExit": true,
    "privacy.sanitize.migrateFx3Prefs": true,
    "security.sandbox.content.tempDirSuffix": "{05adf1fe-b197-4c35-8556-67da79429c71}",
    "storage.vacuum.last.places.sqlite": 1483684196,
    "storage.vacuum.last.index": 0
  },
  "lockedPreferences": {},
  "javaScript": {
    "incrementalGCEnabled": true
  },
  "accessibility": {
    "isActive": false,
    "forceDisabled": 0
  },
  "libraryVersions": {
    "NSPR": {
      "minVersion": "4.14 Beta",
      "version": "4.14 Beta"
    },
    "NSS": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSUTIL": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSSSL": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSSMIME": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    }
  },
  "userJS": {
    "exists": false
  },
  "crashes": {
    "submitted": [],
    "pending": 0
  },
  "sandbox": {
    "contentSandboxLevel": 2
  },
  "experiments": [
    {
      "id": "intersection-observer-nightly5455@experiments.mozilla.org",
      "name": "Intersection Observer API",
      "description": "Measures the impact of enabling the Intersection Observer API in Firefox",
      "active": false,
      "endDate": 1490349779775,
      "detailURL": "",
      "branch": null
    },
    {
      "id": "gpu-process-nightly53@experiments.mozilla.org",
      "name": "GPU Process on Windows",
      "description": "Measures the impact of enabling GPU Process in Firefox Nightly on Windows",
      "active": false,
      "endDate": 1484036951550,
      "detailURL": "",
      "branch": "control"
    }
  ],
  "extensions": [],
  "features": [
    {
      "name": "Activity Stream",
      "version": "0.0.0",
      "id": "activity-stream@mozilla.org"
    },
    {
      "name": "Application Update Service Helper",
      "version": "2.0",
      "id": "aushelper@mozilla.org"
    },
    {
      "name": "FlyWeb",
      "version": "1.0.0",
      "id": "flyweb@mozilla.org"
    },
    {
      "name": "Form Autofill",
      "version": "1.0",
      "id": "formautofill@mozilla.org"
    },
    {
      "name": "Multi-process staged rollout",
      "version": "1.12",
      "id": "e10srollout@mozilla.org"
    },
    {
      "name": "Pocket",
      "version": "1.0.5",
      "id": "firefox@getpocket.com"
    },
    {
      "name": "Presentation",
      "version": "1.0.0",
      "id": "presentation@mozilla.org"
    },
    {
      "name": "Shield Recipe Client",
      "version": "1.0.0",
      "id": "shield-recipe-client@mozilla.org"
    },
    {
      "name": "Site Deployment Checker",
      "version": "1.0",
      "id": "deployment-checker@mozilla.org"
    },
    {
      "name": "Web Compat",
      "version": "1.1",
      "id": "webcompat@mozilla.org"
    },
    {
      "name": "WebCompat Reporter",
      "version": "1.0.0",
      "id": "webcompat-reporter@mozilla.org"
    }
  ],
  "graphics": {
    "numTotalWindows": 1,
    "numAcceleratedWindows": 1,
    "windowLayerManagerType": "Direct3D 11",
    "windowLayerManagerRemote": true,
    "currentAudioBackend": "wasapi",
    "adapterDescription": "NVIDIA GeForce GT 630",
    "adapterVendorID": "0x10de",
    "adapterDeviceID": "0x0f00",
    "adapterSubsysID": "00000000",
    "adapterRAM": "2048",
    "adapterDrivers": "nvd3dumx,nvwgf2umx,nvwgf2umx nvd3dum,nvwgf2um,nvwgf2um",
    "driverVersion": "9.18.13.2049",
    "driverDate": "6-21-2013",
    "adapterDescription2": "",
    "adapterVendorID2": "",
    "adapterDeviceID2": "",
    "adapterSubsysID2": "",
    "adapterRAM2": "",
    "adapterDrivers2": "",
    "driverVersion2": "",
    "driverDate2": "",
    "isGPU2Active": false,
    "direct2DEnabled": true,
    "directWriteEnabled": true,
    "directWriteVersion": "6.2.9200.22082",
    "webgl1Renderer": "Google Inc. -- ANGLE (NVIDIA GeForce GT 630 Direct3D11 vs_5_0 ps_5_0)",
    "webgl1Version": "OpenGL ES 2.0 (ANGLE 2.1.0.2a250c8a0e15)",
    "webgl1DriverExtensions": "GL_ANGLE_depth_texture GL_ANGLE_framebuffer_blit GL_ANGLE_framebuffer_multisample GL_ANGLE_instanced_arrays GL_ANGLE_lossy_etc_decode GL_ANGLE_pack_reverse_row_order GL_ANGLE_robust_client_memory GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ANGLE_texture_usage GL_ANGLE_translated_shader_source GL_CHROMIUM_bind_generates_resource GL_CHROMIUM_bind_uniform_location GL_CHROMIUM_copy_compressed_texture GL_CHROMIUM_copy_texture GL_CHROMIUM_sync_query GL_EXT_blend_minmax GL_EXT_color_buffer_half_float GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_disjoint_timer_query GL_EXT_draw_buffers GL_EXT_frag_depth GL_EXT_map_buffer_range GL_EXT_occlusion_query_boolean GL_EXT_read_format_bgra GL_EXT_robustness GL_EXT_sRGB GL_EXT_shader_texture_lod GL_EXT_texture_compression_dxt1 GL_EXT_texture_filter_anisotropic GL_EXT_texture_format_BGRA8888 GL_EXT_texture_rg GL_EXT_texture_storage GL_EXT_unpack_subimage GL_KHR_debug GL_NV_EGL_stream_consumer_external GL_NV_fence GL_NV_pack_subimage GL_NV_pixel_buffer_object GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth32 GL_OES_element_index_uint GL_OES_get_program_binary GL_OES_mapbuffer GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_npot GL_OES_vertex_array_object ",
    "webgl1Extensions": "ANGLE_instanced_arrays EXT_blend_minmax EXT_color_buffer_half_float EXT_frag_depth EXT_shader_texture_lod EXT_texture_filter_anisotropic EXT_disjoint_timer_query MOZ_debug_get OES_element_index_uint OES_standard_derivatives OES_texture_float OES_texture_float_linear OES_texture_half_float OES_texture_half_float_linear OES_vertex_array_object WEBGL_color_buffer_float WEBGL_compressed_texture_s3tc WEBGL_debug_renderer_info WEBGL_debug_shaders WEBGL_depth_texture WEBGL_draw_buffers WEBGL_lose_context MOZ_WEBGL_lose_context MOZ_WEBGL_compressed_texture_s3tc MOZ_WEBGL_depth_texture",
    "webgl1WSIInfo": "EGL_VENDOR: Google Inc. (adapter LUID: 000000000000a708)\nEGL_VERSION: 1.4 (ANGLE 2.1.0.2a250c8a0e15)\nEGL_EXTENSIONS: EGL_EXT_create_context_robustness EGL_ANGLE_d3d_share_handle_client_buffer EGL_ANGLE_d3d_texture_client_buffer EGL_ANGLE_surface_d3d_texture_2d_share_handle EGL_ANGLE_query_surface_pointer EGL_ANGLE_window_fixed_size EGL_ANGLE_keyed_mutex EGL_ANGLE_surface_orientation EGL_NV_post_sub_buffer EGL_KHR_create_context EGL_EXT_device_query EGL_KHR_image EGL_KHR_image_base EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_get_all_proc_addresses EGL_KHR_stream EGL_KHR_stream_consumer_gltexture EGL_NV_stream_consumer_gltexture_yuv EGL_ANGLE_flexible_surface_compatibility EGL_ANGLE_create_context_webgl_compatibility EGL_CHROMIUM_create_context_bind_generates_resource \nEGL_EXTENSIONS(nullptr): EGL_EXT_client_extensions EGL_EXT_platform_base EGL_EXT_platform_device EGL_ANGLE_platform_angle EGL_ANGLE_platform_angle_d3d EGL_ANGLE_device_creation EGL_ANGLE_device_creation_d3d11 EGL_ANGLE_experimental_present_path EGL_KHR_client_get_all_proc_addresses ",
    "webgl2Renderer": "Google Inc. -- ANGLE (NVIDIA GeForce GT 630 Direct3D11 vs_5_0 ps_5_0)",
    "webgl2Version": "OpenGL ES 3.0 (ANGLE 2.1.0.2a250c8a0e15)",
    "webgl2DriverExtensions": "GL_ANGLE_depth_texture GL_ANGLE_framebuffer_blit GL_ANGLE_framebuffer_multisample GL_ANGLE_instanced_arrays GL_ANGLE_lossy_etc_decode GL_ANGLE_pack_reverse_row_order GL_ANGLE_robust_client_memory GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ANGLE_texture_usage GL_ANGLE_translated_shader_source GL_CHROMIUM_bind_generates_resource GL_CHROMIUM_bind_uniform_location GL_CHROMIUM_copy_compressed_texture GL_CHROMIUM_copy_texture GL_CHROMIUM_sync_query GL_EXT_blend_minmax GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_disjoint_timer_query GL_EXT_draw_buffers GL_EXT_frag_depth GL_EXT_map_buffer_range GL_EXT_occlusion_query_boolean GL_EXT_read_format_bgra GL_EXT_robustness GL_EXT_sRGB GL_EXT_shader_texture_lod GL_EXT_texture_compression_dxt1 GL_EXT_texture_filter_anisotropic GL_EXT_texture_format_BGRA8888 GL_EXT_texture_norm16 GL_EXT_texture_rg GL_EXT_texture_storage GL_EXT_unpack_subimage GL_KHR_debug GL_NV_EGL_stream_consumer_external GL_NV_fence GL_NV_pack_subimage GL_NV_pixel_buffer_object GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_image_external_essl3 GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth32 GL_OES_element_index_uint GL_OES_get_program_binary GL_OES_mapbuffer GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_npot GL_OES_vertex_array_object ",
    "webgl2Extensions": "EXT_color_buffer_float EXT_texture_filter_anisotropic EXT_disjoint_timer_query MOZ_debug_get OES_texture_float_linear WEBGL_compressed_texture_s3tc WEBGL_debug_renderer_info WEBGL_debug_shaders WEBGL_lose_context MOZ_WEBGL_lose_context MOZ_WEBGL_compressed_texture_s3tc",
    "webgl2WSIInfo": "EGL_VENDOR: Google Inc. (adapter LUID: 000000000000a708)\nEGL_VERSION: 1.4 (ANGLE 2.1.0.2a250c8a0e15)\nEGL_EXTENSIONS: EGL_EXT_create_context_robustness EGL_ANGLE_d3d_share_handle_client_buffer EGL_ANGLE_d3d_texture_client_buffer EGL_ANGLE_surface_d3d_texture_2d_share_handle EGL_ANGLE_query_surface_pointer EGL_ANGLE_window_fixed_size EGL_ANGLE_keyed_mutex EGL_ANGLE_surface_orientation EGL_NV_post_sub_buffer EGL_KHR_create_context EGL_EXT_device_query EGL_KHR_image EGL_KHR_image_base EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_get_all_proc_addresses EGL_KHR_stream EGL_KHR_stream_consumer_gltexture EGL_NV_stream_consumer_gltexture_yuv EGL_ANGLE_flexible_surface_compatibility EGL_ANGLE_create_context_webgl_compatibility EGL_CHROMIUM_create_context_bind_generates_resource \nEGL_EXTENSIONS(nullptr): EGL_EXT_client_extensions EGL_EXT_platform_base EGL_EXT_platform_device EGL_ANGLE_platform_angle EGL_ANGLE_platform_angle_d3d EGL_ANGLE_device_creation EGL_ANGLE_device_creation_d3d11 EGL_ANGLE_experimental_present_path EGL_KHR_client_get_all_proc_addresses ",
    "info": {
      "AzureCanvasBackend (UI Process)": "skia",
      "AzureFallbackCanvasBackend (UI Process)": "cairo",
      "AzureContentBackend (UI Process)": "skia",
      "AzureCanvasBackend": "Direct2D 1.1",
      "AzureContentBackend": "Direct2D 1.1",
      "AzureCanvasAccelerated": 0,
      "ApzWheelInput": 1,
      "ApzDragInput": 1
    },
    "featureLog": {
      "features": [
        {
          "name": "HW_COMPOSITING",
          "description": "Compositing",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "D3D11_COMPOSITING",
          "description": "Direct3D11 Compositing",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "DIRECT2D",
          "description": "Direct2D",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "D3D11_HW_ANGLE",
          "description": "Direct3D11 hardware ANGLE",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "GPU_PROCESS",
          "description": "GPU Process",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "WEBRENDER",
          "description": "WebRender",
          "status": "unavailable",
          "log": [
            {
              "type": "default",
              "status": "available"
            },
            {
              "type": "runtime",
              "status": "unavailable",
              "message": "Build doesn't include WebRender"
            }
          ]
        }
      ],
      "fallbacks": []
    },
    "crashGuards": []
  }
}
```

Ubuntu Machine
```json
{
  "application": {
    "name": "Firefox",
    "osVersion": "Linux 3.16.0-77-generic",
    "version": "55.0a1",
    "buildID": "20170309110215",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:55.0) Gecko/20100101 Firefox/55.0",
    "safeMode": false,
    "updateChannel": "nightly",
    "supportURL": "https://support.mozilla.org/1/firefox/55.0a1/Linux/en-US/",
    "numTotalWindows": 1,
    "numRemoteWindows": 1,
    "remoteAutoStart": true,
    "autoStartStatus": 1,
    "keyGoogleFound": true,
    "keyMozillaFound": true
  },
  "modifiedPreferences": {
    "browser.cache.disk.smart_size.first_run": false,
    "browser.cache.disk.hashstats_reported": 1,
    "browser.cache.disk.filesystem_reported": 1,
    "browser.cache.disk.capacity": 358400,
    "browser.cache.frecency_experiment": 2,
    "browser.download.importedFromSqlite": true,
    "browser.download.useDownloadDir": false,
    "browser.places.smartBookmarksVersion": 8,
    "browser.sessionstore.upgradeBackup.latestBuildID": "20170309110215",
    "browser.sessionstore.resume_from_crash": false,
    "browser.sessionstore.restore_on_demand": false,
    "browser.startup.homepage": "about:blank",
    "browser.startup.homepage_override.buildID": "20170309110215",
    "browser.startup.homepage_override.mstone": "55.0a1",
    "browser.urlbar.lastSuggestionsPromptDate": 20161116,
    "browser.urlbar.daysBeforeHidingSuggestionsPrompt": 0,
    "browser.urlbar.usepreloadedtopurls.enabled": false,
    "dom.gamepad.extensions.enabled": true,
    "extensions.lastAppVersion": "55.0a1",
    "javascript.options.asyncstack": false,
    "media.gmp-gmpopenh264.lastUpdate": 1478596309,
    "media.gmp-manager.lastCheck": 1489375704,
    "media.gmp-gmpopenh264.version": "1.6",
    "media.gmp-gmpopenh264.abi": "x86_64-gcc3",
    "media.gmp.storage.version.observed": 1,
    "media.gmp-manager.buildID": "20170309110215",
    "media.decoder-doctor.MediaUnsupportedLibavcodec.formats": "audio/mp4; codecs=\"mp4a.40.2\", audio/mp4; codecs=\"mp4a.40.2\", audio/mp4; codecs=\"mp4a.40.2\", audio/mp4; codecs=\"mp4a.40.2\", audio/mp4; codecs=\"mp4a.40.2\", audio/mp4; codecs=\"mp4a.40.2\", audio/mp4; codecs=\"mp4a.40.2\", audio/mp4; codecs=\"mp4a.40.2\", video/mp4; codecs=\"avc1.4D401E mp4a.40.2\", video/mp4; codecs=\"avc1.42E01E mp4a.40.2\", video/mp4; codecs=\"avc1.4D401E mp4a.40.2\", video/mp4; codecs=\"avc1.4D401F mp4a.40.2\", video/mp4; codecs=\"avc1.640032 mp4a.40.2\", video/mp4; codecs=\"avc1.640033 mp4a.40.2\", video/mp4; codecs=\"avc1.4D401E mp4a.40.2\", video/mp4; codecs=\"avc1.4D401E mp4a.40.2\", video/mp4; codecs=\"avc1.42E01E mp4a.40.2\", video/mp4; codecs=\"avc1.4D401E mp4a.40.2\", video/mp4; codecs=\"avc1.4D401F mp4a.40.2\", video/mp4; codecs=\"avc1.640032 mp4a.40.2\", video/mp4; codecs=\"avc1.640033 mp4a.40.2\", video/mp4; codecs=\"avc1.4D401E mp4a.40.2\", video/mp4; codecs=\"avc1.42E01E mp4a.40.2\", video/mp4; codecs=\"avc1.4D401E mp4a.40.2\", video/mp4; codecs=\"avc1.4D401F mp4a.40.2\", video/mp4; codecs=\"avc1.640032 mp4a.40.2\", video/mp4; codecs=\"avc1.640033 mp4a.40.2\", video/mp4, audio/mp4; codecs=\"mp4a.40.34\", video/mp4; codecs=\"avc1.4d401e\", audio/mp4; codecs=\"mp4a.40.34\", video/mp4; codecs=\"avc1.4d401e\"",
    "network.predictor.cleaned-up": true,
    "network.cookie.prefsMigrated": true,
    "places.database.lastMaintenance": 1478512139,
    "places.history.expiration.transient_current_max_pages": 96580,
    "plugin.importedState": true,
    "plugin.disable_full_page_plugin_for_types": "application/pdf",
    "plugin.state.flash": 0,
    "privacy.sanitize.migrateFx3Prefs": true,
    "storage.vacuum.last.index": 0,
    "storage.vacuum.last.places.sqlite": 1478512139
  },
  "lockedPreferences": {},
  "javaScript": {
    "incrementalGCEnabled": true
  },
  "accessibility": {
    "isActive": false,
    "forceDisabled": 0
  },
  "libraryVersions": {
    "NSPR": {
      "minVersion": "4.13.1",
      "version": "4.13.1"
    },
    "NSS": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSUTIL": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSSSL": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSSMIME": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    }
  },
  "userJS": {
    "exists": false
  },
  "crashes": {
    "submitted": [],
    "pending": 0
  },
  "sandbox": {
    "hasSeccompBPF": true,
    "hasSeccompTSync": true,
    "hasPrivilegedUserNamespaces": true,
    "hasUserNamespaces": true,
    "canSandboxContent": true,
    "canSandboxMedia": true,
    "syscallLog": [],
    "contentSandboxLevel": 2
  },
  "experiments": [],
  "extensions": [
    {
      "name": "Application Update Service Helper",
      "version": "2.0",
      "isActive": true,
      "id": "aushelper@mozilla.org"
    },
    {
      "name": "FlyWeb",
      "version": "1.0.0",
      "isActive": true,
      "id": "flyweb@mozilla.org"
    },
    {
      "name": "Form Autofill",
      "version": "1.0",
      "isActive": true,
      "id": "formautofill@mozilla.org"
    },
    {
      "name": "Multi-process staged rollout",
      "version": "1.11",
      "isActive": true,
      "id": "e10srollout@mozilla.org"
    },
    {
      "name": "Pocket",
      "version": "1.0.5",
      "isActive": true,
      "id": "firefox@getpocket.com"
    },
    {
      "name": "Presentation",
      "version": "1.0.0",
      "isActive": true,
      "id": "presentation@mozilla.org"
    },
    {
      "name": "Shield Recipe Client",
      "version": "1.0.0",
      "isActive": true,
      "id": "shield-recipe-client@mozilla.org"
    },
    {
      "name": "Ubuntu Modifications",
      "version": "3.2",
      "isActive": true,
      "id": "ubufox@ubuntu.com"
    },
    {
      "name": "Web Compat",
      "version": "1.1",
      "isActive": true,
      "id": "webcompat@mozilla.org"
    },
    {
      "name": "WebCompat Reporter",
      "version": "1.0.0",
      "isActive": true,
      "id": "webcompat-reporter@mozilla.org"
    }
  ],
  "graphics": {
    "numTotalWindows": 1,
    "numAcceleratedWindows": 0,
    "windowLayerManagerType": "Basic",
    "windowLayerManagerRemote": true,
    "currentAudioBackend": "pulse",
    "numAcceleratedWindowsMessage": [
      ""
    ],
    "adapterDescription": "nouveau -- Gallium 0.4 on NVC1",
    "adapterVendorID": "nouveau",
    "adapterDeviceID": "Gallium 0.4 on NVC1",
    "adapterRAM": "",
    "adapterDrivers": "",
    "driverVersion": "3.0 Mesa 10.3.2",
    "driverDate": "",
    "webgl1Renderer": "nouveau -- Gallium 0.4 on NVC1",
    "webgl1Version": "3.0 Mesa 10.3.2",
    "webgl1DriverExtensions": "GL_ARB_multisample GL_EXT_abgr GL_EXT_bgra GL_EXT_blend_color GL_EXT_blend_minmax GL_EXT_blend_subtract GL_EXT_copy_texture GL_EXT_polygon_offset GL_EXT_subtexture GL_EXT_texture_object GL_EXT_vertex_array GL_EXT_compiled_vertex_array GL_EXT_texture GL_EXT_texture3D GL_IBM_rasterpos_clip GL_ARB_point_parameters GL_EXT_draw_range_elements GL_EXT_packed_pixels GL_EXT_point_parameters GL_EXT_rescale_normal GL_EXT_separate_specular_color GL_EXT_texture_edge_clamp GL_SGIS_generate_mipmap GL_SGIS_texture_border_clamp GL_SGIS_texture_edge_clamp GL_SGIS_texture_lod GL_ARB_framebuffer_sRGB GL_ARB_multitexture GL_EXT_framebuffer_sRGB GL_IBM_multimode_draw_arrays GL_IBM_texture_mirrored_repeat GL_ARB_texture_cube_map GL_ARB_texture_env_add GL_ARB_transpose_matrix GL_EXT_blend_func_separate GL_EXT_fog_coord GL_EXT_multi_draw_arrays GL_EXT_secondary_color GL_EXT_texture_env_add GL_EXT_texture_filter_anisotropic GL_EXT_texture_lod_bias GL_INGR_blend_func_separate GL_NV_blend_square GL_NV_light_max_exponent GL_NV_texgen_reflection GL_NV_texture_env_combine4 GL_S3_s3tc GL_SUN_multi_draw_arrays GL_ARB_texture_border_clamp GL_ARB_texture_compression GL_EXT_framebuffer_object GL_EXT_texture_compression_s3tc GL_EXT_texture_env_combine GL_EXT_texture_env_dot3 GL_MESA_window_pos GL_NV_packed_depth_stencil GL_NV_texture_rectangle GL_ARB_depth_texture GL_ARB_occlusion_query GL_ARB_shadow GL_ARB_texture_env_combine GL_ARB_texture_env_crossbar GL_ARB_texture_env_dot3 GL_ARB_texture_mirrored_repeat GL_ARB_window_pos GL_EXT_stencil_two_side GL_EXT_texture_cube_map GL_NV_depth_clamp GL_NV_fog_distance GL_APPLE_packed_pixels GL_APPLE_vertex_array_object GL_ARB_draw_buffers GL_ARB_fragment_program GL_ARB_fragment_shader GL_ARB_shader_objects GL_ARB_vertex_program GL_ARB_vertex_shader GL_ATI_draw_buffers GL_ATI_texture_env_combine3 GL_ATI_texture_float GL_EXT_shadow_funcs GL_EXT_stencil_wrap GL_MESA_pack_invert GL_NV_primitive_restart GL_ARB_depth_clamp GL_ARB_fragment_program_shadow GL_ARB_half_float_pixel GL_ARB_occlusion_query2 GL_ARB_point_sprite GL_ARB_shading_language_100 GL_ARB_sync GL_ARB_texture_non_power_of_two GL_ARB_vertex_buffer_object GL_ATI_blend_equation_separate GL_EXT_blend_equation_separate GL_OES_read_format GL_ARB_color_buffer_float GL_ARB_pixel_buffer_object GL_ARB_texture_compression_rgtc GL_ARB_texture_float GL_ARB_texture_rectangle GL_ATI_texture_compression_3dc GL_EXT_packed_float GL_EXT_pixel_buffer_object GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_mirror_clamp GL_EXT_texture_rectangle GL_EXT_texture_sRGB GL_EXT_texture_shared_exponent GL_ARB_framebuffer_object GL_EXT_framebuffer_blit GL_EXT_framebuffer_multisample GL_EXT_packed_depth_stencil GL_ARB_vertex_array_object GL_ATI_separate_stencil GL_ATI_texture_mirror_once GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_gpu_program_parameters GL_EXT_texture_array GL_EXT_texture_compression_latc GL_EXT_texture_integer GL_EXT_texture_sRGB_decode GL_EXT_timer_query GL_OES_EGL_image GL_ARB_copy_buffer GL_ARB_depth_buffer_float GL_ARB_draw_instanced GL_ARB_half_float_vertex GL_ARB_instanced_arrays GL_ARB_map_buffer_range GL_ARB_texture_rg GL_ARB_texture_swizzle GL_ARB_vertex_array_bgra GL_EXT_texture_swizzle GL_EXT_vertex_array_bgra GL_NV_conditional_render GL_AMD_conservative_depth GL_AMD_draw_buffers_blend GL_ARB_ES2_compatibility GL_ARB_blend_func_extended GL_ARB_debug_output GL_ARB_draw_buffers_blend GL_ARB_draw_elements_base_vertex GL_ARB_explicit_attrib_location GL_ARB_fragment_coord_conventions GL_ARB_provoking_vertex GL_ARB_sample_shading GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_shader_texture_lod GL_ARB_texture_cube_map_array GL_ARB_texture_gather GL_ARB_texture_multisample GL_ARB_texture_query_lod GL_ARB_texture_rgb10_a2ui GL_ARB_uniform_buffer_object GL_ARB_vertex_type_2_10_10_10_rev GL_EXT_provoking_vertex GL_EXT_texture_snorm GL_MESA_texture_signed_rgba GL_NV_texture_barrier GL_ARB_get_program_binary GL_ARB_robustness GL_ARB_separate_shader_objects GL_ARB_shader_bit_encoding GL_ARB_texture_compression_bptc GL_ARB_timer_query GL_ARB_transform_feedback2 GL_ARB_transform_feedback3 GL_NV_vdpau_interop GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ARB_base_instance GL_ARB_compressed_texture_pixel_storage GL_ARB_conservative_depth GL_ARB_internalformat_query GL_ARB_map_buffer_alignment GL_ARB_shading_language_420pack GL_ARB_shading_language_packing GL_ARB_texture_storage GL_ARB_transform_feedback_instanced GL_EXT_framebuffer_multisample_blit_scaled GL_EXT_transform_feedback GL_AMD_shader_trinary_minmax GL_ARB_ES3_compatibility GL_ARB_clear_buffer_object GL_ARB_explicit_uniform_location GL_ARB_invalidate_subdata GL_ARB_stencil_texturing GL_ARB_texture_query_levels GL_ARB_texture_storage_multisample GL_ARB_vertex_attrib_binding GL_KHR_debug GL_ARB_buffer_storage GL_ARB_multi_bind GL_ARB_texture_mirror_clamp_to_edge GL_ARB_vertex_type_10f_11f_11f_rev GL_EXT_shader_integer_mix GL_ARB_conditional_render_inverted GL_ARB_derivative_control GL_ARB_texture_barrier ",
    "webgl1Extensions": "ANGLE_instanced_arrays EXT_blend_minmax EXT_color_buffer_half_float EXT_frag_depth EXT_sRGB EXT_shader_texture_lod EXT_texture_filter_anisotropic EXT_disjoint_timer_query MOZ_debug_get OES_element_index_uint OES_standard_derivatives OES_texture_float OES_texture_float_linear OES_texture_half_float OES_texture_half_float_linear OES_vertex_array_object WEBGL_color_buffer_float WEBGL_compressed_texture_etc WEBGL_compressed_texture_s3tc WEBGL_debug_renderer_info WEBGL_debug_shaders WEBGL_depth_texture WEBGL_draw_buffers WEBGL_lose_context MOZ_WEBGL_lose_context MOZ_WEBGL_compressed_texture_s3tc MOZ_WEBGL_depth_texture",
    "webgl1WSIInfo": "GLX 1.4\nGLX_VENDOR(client): Mesa Project and SGI\nGLX_VENDOR(server): SGI\nExtensions: GLX_ARB_create_context GLX_ARB_create_context_profile GLX_ARB_fbconfig_float GLX_ARB_framebuffer_sRGB GLX_ARB_get_proc_address GLX_ARB_multisample GLX_EXT_import_context GLX_EXT_visual_info GLX_EXT_visual_rating GLX_EXT_framebuffer_sRGB GLX_EXT_create_context_es2_profile GLX_MESA_copy_sub_buffer GLX_MESA_multithread_makecurrent GLX_MESA_query_renderer GLX_MESA_swap_control GLX_OML_swap_method GLX_OML_sync_control GLX_SGI_make_current_read GLX_SGI_swap_control GLX_SGI_video_sync GLX_SGIS_multisample GLX_SGIX_fbconfig GLX_SGIX_pbuffer GLX_SGIX_visual_select_group GLX_EXT_texture_from_pixmap GLX_INTEL_swap_event ",
    "webgl2Renderer": "nouveau -- Gallium 0.4 on NVC1",
    "webgl2Version": "3.3 (Core Profile) Mesa 10.3.2",
    "webgl2DriverExtensions": "GL_ARB_ES2_compatibility GL_ARB_ES3_compatibility GL_ARB_base_instance GL_ARB_blend_func_extended GL_ARB_buffer_storage GL_ARB_clear_buffer_object GL_ARB_color_buffer_float GL_ARB_compressed_texture_pixel_storage GL_ARB_conditional_render_inverted GL_ARB_copy_buffer GL_ARB_conservative_depth GL_ARB_debug_output GL_ARB_depth_buffer_float GL_ARB_depth_clamp GL_ARB_derivative_control GL_ARB_draw_buffers GL_ARB_draw_buffers_blend GL_ARB_draw_elements_base_vertex GL_ARB_draw_indirect GL_ARB_draw_instanced GL_ARB_explicit_attrib_location GL_ARB_explicit_uniform_location GL_ARB_fragment_coord_conventions GL_ARB_fragment_layer_viewport GL_ARB_fragment_shader GL_ARB_framebuffer_object GL_ARB_framebuffer_sRGB GL_ARB_get_program_binary GL_ARB_gpu_shader5 GL_ARB_half_float_pixel GL_ARB_half_float_vertex GL_ARB_instanced_arrays GL_ARB_internalformat_query GL_ARB_invalidate_subdata GL_ARB_map_buffer_alignment GL_ARB_map_buffer_range GL_ARB_multi_bind GL_ARB_multi_draw_indirect GL_ARB_occlusion_query2 GL_ARB_pixel_buffer_object GL_ARB_point_sprite GL_ARB_provoking_vertex GL_ARB_robustness GL_ARB_sample_shading GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_separate_shader_objects GL_ARB_shader_bit_encoding GL_ARB_shader_objects GL_ARB_shader_texture_lod GL_ARB_shading_language_packing GL_ARB_shading_language_420pack GL_ARB_stencil_texturing GL_ARB_sync GL_ARB_texture_barrier GL_ARB_texture_buffer_object GL_ARB_texture_buffer_object_rgb32 GL_ARB_texture_buffer_range GL_ARB_texture_compression_bptc GL_ARB_texture_compression_rgtc GL_ARB_texture_cube_map_array GL_ARB_texture_float GL_ARB_texture_gather GL_ARB_texture_mirror_clamp_to_edge GL_ARB_texture_multisample GL_ARB_texture_non_power_of_two GL_ARB_texture_query_levels GL_ARB_texture_query_lod GL_ARB_texture_rectangle GL_ARB_texture_rgb10_a2ui GL_ARB_texture_rg GL_ARB_texture_storage GL_ARB_texture_storage_multisample GL_ARB_texture_swizzle GL_ARB_timer_query GL_ARB_transform_feedback2 GL_ARB_transform_feedback3 GL_ARB_transform_feedback_instanced GL_ARB_uniform_buffer_object GL_ARB_vertex_array_bgra GL_ARB_vertex_array_object GL_ARB_vertex_attrib_binding GL_ARB_vertex_shader GL_ARB_vertex_type_10f_11f_11f_rev GL_ARB_vertex_type_2_10_10_10_rev GL_ARB_viewport_array GL_EXT_abgr GL_EXT_blend_equation_separate GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_framebuffer_blit GL_EXT_framebuffer_multisample GL_EXT_framebuffer_multisample_blit_scaled GL_EXT_framebuffer_sRGB GL_EXT_packed_depth_stencil GL_EXT_packed_float GL_EXT_pixel_buffer_object GL_EXT_provoking_vertex GL_EXT_shader_integer_mix GL_EXT_texture_array GL_EXT_texture_compression_dxt1 GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_EXT_texture_compression_latc GL_EXT_texture_compression_rgtc GL_EXT_texture_compression_s3tc GL_EXT_texture_filter_anisotropic GL_EXT_texture_integer GL_EXT_texture_mirror_clamp GL_EXT_texture_shared_exponent GL_EXT_texture_snorm GL_EXT_texture_sRGB GL_EXT_texture_sRGB_decode GL_EXT_texture_swizzle GL_EXT_timer_query GL_EXT_transform_feedback GL_EXT_vertex_array_bgra GL_OES_EGL_image GL_OES_read_format GL_KHR_debug GL_AMD_conservative_depth GL_AMD_draw_buffers_blend GL_AMD_shader_trinary_minmax GL_ATI_blend_equation_separate GL_ATI_texture_compression_3dc GL_ATI_texture_float GL_ATI_texture_mirror_once GL_IBM_multimode_draw_arrays GL_MESA_pack_invert GL_MESA_texture_signed_rgba GL_NV_conditional_render GL_NV_depth_clamp GL_NV_packed_depth_stencil GL_NV_texture_barrier GL_NV_vdpau_interop GL_S3_s3tc",
    "webgl2Extensions": "EXT_color_buffer_float EXT_texture_filter_anisotropic EXT_disjoint_timer_query MOZ_debug_get OES_texture_float_linear WEBGL_compressed_texture_etc WEBGL_compressed_texture_s3tc WEBGL_debug_renderer_info WEBGL_debug_shaders WEBGL_lose_context MOZ_WEBGL_lose_context MOZ_WEBGL_compressed_texture_s3tc",
    "webgl2WSIInfo": "GLX 1.4\nGLX_VENDOR(client): Mesa Project and SGI\nGLX_VENDOR(server): SGI\nExtensions: GLX_ARB_create_context GLX_ARB_create_context_profile GLX_ARB_fbconfig_float GLX_ARB_framebuffer_sRGB GLX_ARB_get_proc_address GLX_ARB_multisample GLX_EXT_import_context GLX_EXT_visual_info GLX_EXT_visual_rating GLX_EXT_framebuffer_sRGB GLX_EXT_create_context_es2_profile GLX_MESA_copy_sub_buffer GLX_MESA_multithread_makecurrent GLX_MESA_query_renderer GLX_MESA_swap_control GLX_OML_swap_method GLX_OML_sync_control GLX_SGI_make_current_read GLX_SGI_swap_control GLX_SGI_video_sync GLX_SGIS_multisample GLX_SGIX_fbconfig GLX_SGIX_pbuffer GLX_SGIX_visual_select_group GLX_EXT_texture_from_pixmap GLX_INTEL_swap_event ",
    "info": {
      "AzureCanvasBackend": "skia",
      "AzureFallbackCanvasBackend": "none",
      "AzureContentBackend": "skia",
      "AzureCanvasAccelerated": 0,
      "CairoUseXRender": 0,
      "ApzWheelInput": 1,
      "ApzTouchInput": 1,
      "ApzDragInput": 1
    },
    "featureLog": {
      "features": [
        {
          "name": "HW_COMPOSITING",
          "description": "Compositing",
          "status": "blocked",
          "log": [
            {
              "type": "default",
              "status": "blocked",
              "message": "Acceleration blocked by platform"
            }
          ]
        },
        {
          "name": "OPENGL_COMPOSITING",
          "description": "OpenGL Compositing",
          "status": "unavailable",
          "log": [
            {
              "type": "default",
              "status": "unavailable",
              "message": "Hardware compositing is disabled"
            }
          ]
        },
        {
          "name": "WEBRENDER",
          "description": "WebRender",
          "status": "unavailable",
          "log": [
            {
              "type": "default",
              "status": "available"
            },
            {
              "type": "runtime",
              "status": "unavailable",
              "message": "Build doesn't include WebRender"
            }
          ]
        }
      ],
      "fallbacks": []
    },
    "crashGuards": []
  }
}

```

**Comment by MikeLien** [comment]
The Windows test machine's information for FT prototype case as below:

```json
{
  "application": {
    "name": "Firefox",
    "osVersion": "Windows_NT 6.1",
    "version": "54.0a1",
    "buildID": "20170223030204",
    "userAgent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:54.0) Gecko/20100101 Firefox/54.0",
    "safeMode": false,
    "updateChannel": "nightly",
    "supportURL": "https://support.mozilla.org/1/firefox/54.0a1/WINNT/en-US/",
    "numTotalWindows": 1,
    "numRemoteWindows": 1,
    "remoteAutoStart": true,
    "autoStartStatus": 1
  },
  "modifiedPreferences": {
    "accessibility.typeaheadfind.flashBar": 0,
    "accessibility.loadedInLastSession": true,
    "accessibility.lastLoadDate": 1490669074,
    "browser.cache.disk.hashstats_reported": 1,
    "browser.cache.frecency_experiment": 4,
    "browser.cache.disk.smart_size.first_run": false,
    "browser.cache.disk.smart_size.use_old_max": false,
    "browser.cache.disk.capacity": 358400,
    "browser.cache.disk.filesystem_reported": 1,
    "browser.cache.disk.smart_size_cached_value": 358400,
    "browser.download.importedFromSqlite": true,
    "browser.download.manager.alertOnEXEOpen": true,
    "browser.places.importBookmarksHTML": false,
    "browser.places.smartBookmarksVersion": 8,
    "browser.sessionstore.upgradeBackup.latestBuildID": "20170223030204",
    "browser.sessionstore.resume_from_crash": false,
    "browser.sessionstore.restore_on_demand": false,
    "browser.startup.homepage": "about:blank",
    "browser.startup.homepage_override.buildID": "20170223030204",
    "browser.startup.homepage_override.mstone": "54.0a1",
    "browser.tabs.remote.autostart.2": true,
    "browser.tabs.crashReporting.sendReport": false,
    "browser.urlbar.userMadeSearchSuggestionsChoice": true,
    "browser.urlbar.usepreloadedtopurls.enabled": false,
    "dom.apps.lastUpdate.buildID": "20161019084923",
    "dom.apps.reset-permissions": true,
    "dom.gamepad.extensions.enabled": true,
    "dom.apps.lastUpdate.mstone": "49.0.2",
    "dom.disable_open_during_load": false,
    "dom.mozApps.used": true,
    "extensions.lastAppVersion": "54.0a1",
    "font.internaluseonly.changed": true,
    "gfx.crash-guard.status.d3d11layers": 2,
    "gfx.crash-guard.d3d11layers.appVersion": "49.0.2",
    "gfx.driver-init.feature-d3d11": true,
    "gfx.crash-guard.d3d11layers.feature-d3d11": true,
    "gfx.driver-init.deviceID": "0x0f00",
    "gfx.driver-init.driverVersion": "9.18.13.2049",
    "gfx.driver-init.status": 2,
    "gfx.driver-init.appVersion": "42.0",
    "gfx.crash-guard.d3d11layers.feature-d2d": true,
    "gfx.crash-guard.d3d11layers.driverVersion": "9.18.13.2049",
    "gfx.direct3d.last_used_feature_level_idx": 0,
    "gfx.crash-guard.d3d11layers.deviceID": "0x1284",
    "gfx.driver-init.feature-d2d": true,
    "gfx.crash-guard.status.d3d9video": 2,
    "keyword.URL": true,
    "media.gmp-eme-adobe.abi": "x86-msvc-x64",
    "media.gmp.storage.version.observed": 1,
    "media.gmp-eme-adobe.lastUpdate": 1464597454,
    "media.gmp-manager.buildID": "20170223030204",
    "media.webrtc.debug.log_file": "C:\\Users\\user\\AppData\\Local\\Temp\\WebRTC.log",
    "media.benchmark.vp9.versioncheck": 1,
    "media.gmp-gmpopenh264.version": "1.6",
    "media.benchmark.vp9.fps": 220,
    "media.hardware-video-decoding.failed": false,
    "media.gmp-eme-adobe.version": "17",
    "media.gmp-gmpopenh264.abi": "x86_64-msvc-x64",
    "media.gmp-gmpopenh264.lastUpdate": 1490669136,
    "media.gmp-manager.lastCheck": 1490669135,
    "media.gmp-widevinecdm.lastUpdate": 1490669137,
    "media.gmp-widevinecdm.abi": "x86_64-msvc-x64",
    "media.gmp-widevinecdm.version": "1.4.8.903",
    "network.cookie.prefsMigrated": true,
    "network.predictor.cleaned-up": true,
    "places.history.expiration.transient_current_max_pages": 122334,
    "places.database.lastMaintenance": 1490580346,
    "plugin.importedState": true,
    "plugin.disable_full_page_plugin_for_types": "application/pdf",
    "privacy.sanitize.timeSpan": 0,
    "privacy.sanitize.migrateClearSavedPwdsOnExit": true,
    "privacy.cpd.cookies": false,
    "privacy.sanitize.migrateFx3Prefs": true,
    "privacy.cpd.formdata": false,
    "privacy.cpd.cache": false,
    "privacy.cpd.sessions": false,
    "security.sandbox.content.tempDirSuffix": "{d973fb75-1935-4456-9b96-c23d67ad9029}",
    "security.disable_button.openCertManager": false,
    "services.sync.declinedEngines": "",
    "storage.vacuum.last.places.sqlite": 1490580345,
    "storage.vacuum.last.index": 0
  },
  "lockedPreferences": {},
  "javaScript": {
    "incrementalGCEnabled": true
  },
  "accessibility": {
    "isActive": true,
    "forceDisabled": 0
  },
  "libraryVersions": {
    "NSPR": {
      "minVersion": "4.13.1",
      "version": "4.13.1"
    },
    "NSS": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSUTIL": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSSSL": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    },
    "NSSSMIME": {
      "minVersion": "3.30 Beta",
      "version": "3.30 Beta"
    }
  },
  "userJS": {
    "exists": false
  },
  "crashes": {
    "submitted": [],
    "pending": 0
  },
  "sandbox": {
    "contentSandboxLevel": 2
  },
  "experiments": [
    {
      "id": "tls13-compat-nightly52@experiments.mozilla.org",
      "name": "TLS 1.3 Compatibility Testing",
      "description": "Testing compatibility of TLS 1.3.",
      "active": false,
      "endDate": 1479352535093,
      "detailURL": "",
      "branch": null
    }
  ],
  "extensions": [
    {
      "name": "Application Update Service Helper",
      "version": "2.0",
      "isActive": true,
      "id": "aushelper@mozilla.org"
    },
    {
      "name": "FlyWeb",
      "version": "1.0.0",
      "isActive": true,
      "id": "flyweb@mozilla.org"
    },
    {
      "name": "Form Autofill",
      "version": "1.0",
      "isActive": true,
      "id": "formautofill@mozilla.org"
    },
    {
      "name": "Multi-process staged rollout",
      "version": "1.9",
      "isActive": true,
      "id": "e10srollout@mozilla.org"
    },
    {
      "name": "Pocket",
      "version": "1.0.5",
      "isActive": true,
      "id": "firefox@getpocket.com"
    },
    {
      "name": "Presentation",
      "version": "1.0.0",
      "isActive": true,
      "id": "presentation@mozilla.org"
    },
    {
      "name": "SHA-1 deprecation staged rollout",
      "version": "1.0",
      "isActive": true,
      "id": "disableSHA1rollout@mozilla.org"
    },
    {
      "name": "Shield Recipe Client",
      "version": "1.0.0",
      "isActive": true,
      "id": "shield-recipe-client@mozilla.org"
    },
    {
      "name": "Web Compat",
      "version": "1.1",
      "isActive": true,
      "id": "webcompat@mozilla.org"
    },
    {
      "name": "WebCompat Reporter",
      "version": "1.0.0",
      "isActive": true,
      "id": "webcompat-reporter@mozilla.org"
    }
  ],
  "graphics": {
    "numTotalWindows": 1,
    "numAcceleratedWindows": 1,
    "windowLayerManagerType": "Direct3D 11",
    "windowLayerManagerRemote": true,
    "currentAudioBackend": "wasapi",
    "adapterDescription": "NVIDIA GeForce GT 630  ",
    "adapterVendorID": "0x10de",
    "adapterDeviceID": "0x1284",
    "adapterSubsysID": "00000000",
    "adapterRAM": "2048",
    "adapterDrivers": "nvd3dumx,nvwgf2umx,nvwgf2umx nvd3dum,nvwgf2um,nvwgf2um",
    "driverVersion": "9.18.13.2049",
    "driverDate": "6-21-2013",
    "adapterDescription2": "",
    "adapterVendorID2": "",
    "adapterDeviceID2": "",
    "adapterSubsysID2": "",
    "adapterRAM2": "",
    "adapterDrivers2": "",
    "driverVersion2": "",
    "driverDate2": "",
    "isGPU2Active": false,
    "direct2DEnabled": true,
    "directWriteEnabled": true,
    "directWriteVersion": "6.2.9200.17461",
    "webgl1Renderer": "Google Inc. -- ANGLE (NVIDIA GeForce GT 630   Direct3D11 vs_5_0 ps_5_0)",
    "webgl1Version": "OpenGL ES 2.0 (ANGLE 2.1.0.2a250c8a0e15)",
    "webgl1Extensions": "GL_ANGLE_depth_texture GL_ANGLE_framebuffer_blit GL_ANGLE_framebuffer_multisample GL_ANGLE_instanced_arrays GL_ANGLE_lossy_etc_decode GL_ANGLE_pack_reverse_row_order GL_ANGLE_robust_client_memory GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ANGLE_texture_usage GL_ANGLE_translated_shader_source GL_CHROMIUM_bind_generates_resource GL_CHROMIUM_bind_uniform_location GL_CHROMIUM_copy_compressed_texture GL_CHROMIUM_copy_texture GL_CHROMIUM_sync_query GL_EXT_blend_minmax GL_EXT_color_buffer_half_float GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_disjoint_timer_query GL_EXT_draw_buffers GL_EXT_frag_depth GL_EXT_map_buffer_range GL_EXT_occlusion_query_boolean GL_EXT_read_format_bgra GL_EXT_robustness GL_EXT_sRGB GL_EXT_shader_texture_lod GL_EXT_texture_compression_dxt1 GL_EXT_texture_filter_anisotropic GL_EXT_texture_format_BGRA8888 GL_EXT_texture_rg GL_EXT_texture_storage GL_EXT_unpack_subimage GL_KHR_debug GL_NV_EGL_stream_consumer_external GL_NV_fence GL_NV_pack_subimage GL_NV_pixel_buffer_object GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth32 GL_OES_element_index_uint GL_OES_get_program_binary GL_OES_mapbuffer GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_npot GL_OES_vertex_array_object ",
    "webgl1WSIInfo": "EGL_VENDOR: Google Inc. (adapter LUID: 0000000000008283)\nEGL_VERSION: 1.4 (ANGLE 2.1.0.2a250c8a0e15)\nEGL_EXTENSIONS: EGL_EXT_create_context_robustness EGL_ANGLE_d3d_share_handle_client_buffer EGL_ANGLE_d3d_texture_client_buffer EGL_ANGLE_surface_d3d_texture_2d_share_handle EGL_ANGLE_query_surface_pointer EGL_ANGLE_window_fixed_size EGL_ANGLE_keyed_mutex EGL_ANGLE_surface_orientation EGL_NV_post_sub_buffer EGL_KHR_create_context EGL_EXT_device_query EGL_KHR_image EGL_KHR_image_base EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_get_all_proc_addresses EGL_KHR_stream EGL_KHR_stream_consumer_gltexture EGL_NV_stream_consumer_gltexture_yuv EGL_ANGLE_flexible_surface_compatibility EGL_ANGLE_create_context_webgl_compatibility EGL_CHROMIUM_create_context_bind_generates_resource \nEGL_EXTENSIONS(nullptr): EGL_EXT_client_extensions EGL_EXT_platform_base EGL_EXT_platform_device EGL_ANGLE_platform_angle EGL_ANGLE_platform_angle_d3d EGL_ANGLE_device_creation EGL_ANGLE_device_creation_d3d11 EGL_ANGLE_experimental_present_path EGL_KHR_client_get_all_proc_addresses ",
    "webgl2Renderer": "Google Inc. -- ANGLE (NVIDIA GeForce GT 630   Direct3D11 vs_5_0 ps_5_0)",
    "webgl2Version": "OpenGL ES 3.0 (ANGLE 2.1.0.2a250c8a0e15)",
    "webgl2Extensions": "GL_ANGLE_depth_texture GL_ANGLE_framebuffer_blit GL_ANGLE_framebuffer_multisample GL_ANGLE_instanced_arrays GL_ANGLE_lossy_etc_decode GL_ANGLE_pack_reverse_row_order GL_ANGLE_robust_client_memory GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ANGLE_texture_usage GL_ANGLE_translated_shader_source GL_CHROMIUM_bind_generates_resource GL_CHROMIUM_bind_uniform_location GL_CHROMIUM_copy_compressed_texture GL_CHROMIUM_copy_texture GL_CHROMIUM_sync_query GL_EXT_blend_minmax GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_disjoint_timer_query GL_EXT_draw_buffers GL_EXT_frag_depth GL_EXT_map_buffer_range GL_EXT_occlusion_query_boolean GL_EXT_read_format_bgra GL_EXT_robustness GL_EXT_sRGB GL_EXT_shader_texture_lod GL_EXT_texture_compression_dxt1 GL_EXT_texture_filter_anisotropic GL_EXT_texture_format_BGRA8888 GL_EXT_texture_norm16 GL_EXT_texture_rg GL_EXT_texture_storage GL_EXT_unpack_subimage GL_KHR_debug GL_NV_EGL_stream_consumer_external GL_NV_fence GL_NV_pack_subimage GL_NV_pixel_buffer_object GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_image_external_essl3 GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth32 GL_OES_element_index_uint GL_OES_get_program_binary GL_OES_mapbuffer GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_npot GL_OES_vertex_array_object ",
    "webgl2WSIInfo": "EGL_VENDOR: Google Inc. (adapter LUID: 0000000000008283)\nEGL_VERSION: 1.4 (ANGLE 2.1.0.2a250c8a0e15)\nEGL_EXTENSIONS: EGL_EXT_create_context_robustness EGL_ANGLE_d3d_share_handle_client_buffer EGL_ANGLE_d3d_texture_client_buffer EGL_ANGLE_surface_d3d_texture_2d_share_handle EGL_ANGLE_query_surface_pointer EGL_ANGLE_window_fixed_size EGL_ANGLE_keyed_mutex EGL_ANGLE_surface_orientation EGL_NV_post_sub_buffer EGL_KHR_create_context EGL_EXT_device_query EGL_KHR_image EGL_KHR_image_base EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_get_all_proc_addresses EGL_KHR_stream EGL_KHR_stream_consumer_gltexture EGL_NV_stream_consumer_gltexture_yuv EGL_ANGLE_flexible_surface_compatibility EGL_ANGLE_create_context_webgl_compatibility EGL_CHROMIUM_create_context_bind_generates_resource \nEGL_EXTENSIONS(nullptr): EGL_EXT_client_extensions EGL_EXT_platform_base EGL_EXT_platform_device EGL_ANGLE_platform_angle EGL_ANGLE_platform_angle_d3d EGL_ANGLE_device_creation EGL_ANGLE_device_creation_d3d11 EGL_ANGLE_experimental_present_path EGL_KHR_client_get_all_proc_addresses ",
    "info": {
      "AzureCanvasBackend (UI Process)": "skia",
      "AzureFallbackCanvasBackend (UI Process)": "cairo",
      "AzureContentBackend (UI Process)": "skia",
      "AzureCanvasBackend": "Direct2D 1.1",
      "AzureContentBackend": "Direct2D 1.1",
      "AzureCanvasAccelerated": 0,
      "ApzWheelInput": 1,
      "ApzTouchInput": 1,
      "ApzDragInput": 1
    },
    "featureLog": {
      "features": [
        {
          "name": "HW_COMPOSITING",
          "description": "Compositing",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "D3D11_COMPOSITING",
          "description": "Direct3D11 Compositing",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "D3D9_COMPOSITING",
          "description": "Direct3D9 Compositing",
          "status": "disabled",
          "log": [
            {
              "type": "default",
              "status": "disabled",
              "message": "Disabled by default"
            }
          ]
        },
        {
          "name": "DIRECT2D",
          "description": "Direct2D",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "D3D11_HW_ANGLE",
          "description": "Direct3D11 hardware ANGLE",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "GPU_PROCESS",
          "description": "GPU Process",
          "status": "available",
          "log": [
            {
              "type": "default",
              "status": "available"
            }
          ]
        },
        {
          "name": "WEBRENDER",
          "description": "WebRender",
          "status": "unavailable",
          "log": [
            {
              "type": "default",
              "status": "available"
            },
            {
              "type": "runtime",
              "status": "unavailable",
              "message": "Build doesn't include WebRender"
            }
          ]
        }
      ],
      "fallbacks": []
    },
    "crashGuards": []
  }
}
```

**Comment by bobbyc** [comment]
MVP done
